lhotse/cut/set.py

```diff
@@ -3557,10 +3559,9 @@ def noise_gen():
             # Keep sampling until we mixed in a "duration" amount of noise.
             # Note: we subtract 0.05s (50ms) from the target duration to avoid edge cases
             #       where we mix in some noise cut that effectively has 0 frames of features.
-            while mixed_in_duration < (
-                self.duration if self.duration is not None else cut.duration - 0.05
-            ):
+            while mixed_in_duration < target_mixed_duration:
                 to_mix = next(mix_in_cuts)
+                to_mix = self._maybe_truncate_cut(to_mix, target_mixed_duration, rng)
```

Comment:
@pzelasko I just noticed this. We need to update the target duration for random sub-region selection in the `while` loop (like I had in the PR). Let's assume the target is 20s. The first noise cut is 10s long, and we enter the `while` loop. The second noise cut is 15s long so we can select a random sub-region of 10s from the 15s, however because we use the original target duration we select from 0. Do you agree?

Reply:
I think it makes sense. Can you make the change and verify it works as expected in another PR? Ideally add a unit test that would have been broken before the change is made, it will help maintaining this bit of code in the future.

Reply:
Done #1317. Thanks.